### PR TITLE
[feature] [refactor] Add Bing to the list of search engines.

### PR DESCRIPTION
**Pull request by sytone**
Updated the process to upgrade config to reduce boilerplate code
Bumped version to 3.27.0

### CODE REVIEW

**Comment by zombieFox** [comment]
Looks good! A couple of points:

- Is `.vscode/launch.json` needed? Not sure what it is doing. Also looks like an editor specific file. 
- I've caused some conflicts which I'll aim to fix asap.

File: src/js/update.js [hunk 1]
```diff
@@ -49,6 +48,9 @@ var update = (function() {
             giphy: {
               url: "https://giphy.com/search/"
             },
+            bing: {
+              url: "https://www.bing.com/search?q="
+            },
```
Comment (zombieFox) [inline src/js/update.js]: Do we need to add the `bing` object if it is added on line 597 in the `versionUpdates["3.27.0"]` function?

File: src/js/update.js [hunk 2]
```diff
@@ -577,160 +580,192 @@ var update = (function() {
...
       if (version.compare(data.version, "3.21.0") == -1) {
         console.log("\t= running update 3.21.0");
         data = _update_3210(data);
-      };
-    };
+      }
+
+      // Shift to a associate array for the configuration update as
+      // there less code to maintain.
+      for (var key in versionUpdates) {
```
Comment (zombieFox) [inline src/js/update.js]: This looks awesome!

File: src/js/update.js (hunk 1 above)
Comment (zombieFox) [reply src/js/update.js]: Also, I don't think changing previous update functions. Would rather reflect the change in the `state` object with a new update function.

**Comment by zombieFox** [comment]
Fixed all the conflicts ... which I caused. So yea.. good to merge :shipit:

**Comment by sytone** [comment]
Thanks, love the page just wanted to add my favorite search engine. The .vscode allows me to run the chrome debugger in Visual Studio Code. This is a generic configuration that can be shared. I do all editing in VS Code these days.

**Pull request merged**